### PR TITLE
Add support for custom font families and styling to watermarks

### PR DESCRIPTION
**Pull request by meetmangukiya**
Closes https://github.com/tradingview/lightweight-charts/issues/437

**Type of PR:** 

**PR checklist:**

- [x] Addresses an existing issue: fixes https://github.com/tradingview/lightweight-charts/issues/437
- [ ] Includes tests
- [x] Documentation update

**Overview of change:**

**Is there anything you'd like reviewers to focus on?**

### CODE REVIEW

**Comment by meetmangukiya** [comment]
I am not sure how to fix the dts-changes build, any suggestions?

File: src/api/options/watermark-options-defaults.ts [hunk 1]
```diff
@@ -4,6 +4,8 @@ export const watermarkOptionsDefaults: WatermarkOptions = {
 	color: 'rgba(0, 0, 0, 0)',
 	visible: false,
 	fontSize: 48,
+	fontFamily: '',
```
Comment (timocov) [inline src/api/options/watermark-options-defaults.ts:7]: This doesn't fit with default value from the docs. I guess we need to use `defaultFontFamily` from `make-font.ts`?

File: src/views/pane/watermark-pane-view.ts [hunk 2]
```diff
@@ -56,7 +56,7 @@ export class WatermarkPaneView implements IUpdatablePaneView {
 		data.lines = [
 			{
 				text: options.text,
-				font: makeFont(options.fontSize),
+				font: makeFont(options.fontSize, options.fontFamily, options.fontStyle),
```
Comment (timocov) [inline src/views/pane/watermark-pane-view.ts:59]: If `options.fontFamily` isn't `undefined` (default value is empty string?), then `makeFont` will take it as the value (I guess we expect to use default one instead in this case?). I guess we need 1) to change default font family and 2) change `makeFont` so it shouldn't allow empty font family (not sure about that, what do you think?).

**Comment by timocov** [comment]
> I am not sure how to fix the dts-changes build, any suggestions?

That's expected. That test means that we have some changes in public API so we need to carry about that and see that everything is public as expected. In your case this is expected so please don't worry about it.

File: src/api/options/watermark-options-defaults.ts (hunk 1 above)
Comment (meetmangukiya) [reply src/api/options/watermark-options-defaults.ts:7]: right, should've been undefined for the default one to get used: https://github.com/tradingview/lightweight-charts/blob/master/src/helpers/make-font.ts#L18

Comment (timocov) [reply src/api/options/watermark-options-defaults.ts:7]: Yeah, maybe `undefined` is good option as well. https://github.com/tradingview/lightweight-charts/pull/448#discussion_r429868583

**Comment by timocov** [comment]
Hey @meetmangukiya, do you have any updates here? I'm asking because we're going to prepare to 3.0 release soon and I think it'd be awesome to put it into that release as well. If you haven't enough time for it, please let me know and I think we'll finish it.

**Comment by meetmangukiya** [comment]
Hey @timocov sorry for the delay, even though its a small change, haven't been able to find the time to do it and would need some more time to get around this.

**Comment by timocov** [comment]
@meetmangukiya sure, no worries and no problem at all. Just to be sure - you're going to finish it but you need a bit time, right? (to avoid misunderstanding).

**Comment by meetmangukiya** [comment]
@timocov when is the release scheduled for?

**Comment by timocov** [comment]
We don't have specific date for it, but I hope it should be within 1-2 weeks.

**Comment by timocov** [comment]
Hey @meetmangukiya, any updates on this? As I said, we can finish it if you'd like (no pressure, just a reminder 🙂).

**Comment by meetmangukiya** [comment]
@timocov I pushed the changes. Not sure if I got the graphics test right. Also, don't think it will work because this PR introduces visual difference, not sure if it is above or below the tolerance threshold. Sorry for the delay.

**Comment by timocov** [comment]
Hey @meetmangukiya, thanks for update and no worries! I just added some changes ahead of yours, but nothing critical.

> Not sure if I got the graphics test right. 

You got them completely right!

> Also, don't think it will work because this PR introduces visual difference, not sure if it is above or below the tolerance threshold

That's ok. That tests might be broken, they just help us see what's actually changed in visual part.

**Comment by timocov** [comment]
Merged. @meetmangukiya thank you for your contribution!

**Pull request merged**